tests/test_parser_requirements.py

```diff
@@ -17,66 +17,100 @@
 import os
-import unittest
 from unittest import TestCase
 
-from cyclonedx_py.parser.requirements import RequirementsParser
+from cyclonedx_py.parser.requirements import RequirementsParser, RequirementsFileParser
 
 
 class TestRequirementsParser(TestCase):
 
     def test_simple(self) -> None:
-        with open(os.path.join(os.path.dirname(__file__), 'fixtures/requirements-simple.txt')) as r:
+        with open(os.path.join(os.path.dirname(__file__),
+                               'fixtures/requirements-simple.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-            r.close()
         self.assertTrue(1, parser.component_count())
         self.assertFalse(parser.has_warnings())
 
     def test_example_1(self) -> None:
-        with open(os.path.join(os.path.dirname(__file__), 'fixtures/requirements-example-1.txt')) as r:
+        with open(os.path.join(os.path.dirname(__file__),
+                               'fixtures/requirements-example-1.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-            r.close()
         self.assertTrue(3, parser.component_count())
         self.assertFalse(parser.has_warnings())
 
     def test_example_with_comments(self) -> None:
-        with open(os.path.join(os.path.dirname(__file__), 'fixtures/requirements-with-comments.txt')) as r:
+        with open(os.path.join(os.path.dirname(__file__),
+                               'fixtures/requirements-with-comments.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-            r.close()
         self.assertTrue(5, parser.component_count())
         self.assertFalse(parser.has_warnings())
 
     def test_example_multiline_with_comments(self) -> None:
-        with open(os.path.join(os.path.dirname(__file__), 'fixtures/requirements-multilines-with-comments.txt')) as r:
+        with open(os.path.join(os.path.dirname(__file__),
+                               'fixtures/requirements-multilines-with-comments.txt')) as r:
             parser = RequirementsParser(
                 requirements_content=r.read()
             )
-            r.close()
         self.assertTrue(5, parser.component_count())
         self.assertFalse(parser.has_warnings())
 
-    @unittest.skip('Not yet supported')
+    def test_example_local_packages(self) -> None:
+        with open(os.path.join(os.path.dirname(__file__),
+                               'fixtures/requirements-local-and-remote-packages.txt')) as r:
+            parser = RequirementsParser(
+                requirements_content=r.read()
+            )
+        self.assertTrue(6, parser.component_count())
+        self.assertFalse(parser.has_warnings())
+
+    def test_example_local_and_nested_packages(self) -> None:
+        # RequirementsFileParser can parse nested requirements files,
```

Comment:
This is a great feature to note @mostafa - we need to make sure the documentation is updated to reflect this please.

Reply:
I updated the specific section of the docs about the nested files.